library_automations/library/generator/metadata/metadata.go

```diff
@@ -0,0 +1,68 @@
+package metadata
+
+import (
+	"fmt"
+	"github.com/blinkops/blink-sdk/internal/text"
+	"github.com/blinkops/blink-sdk/library_automations/consts"
+	"github.com/blinkops/blink-sdk/library_automations/types"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"sort"
+)
+
+func CreateMetadataObject(actions, resultsAction *types.Action, libraryAutomation *types.Automation) *types.Metadata {
+	return &types.Metadata{
+		ID:                     uuid.NewV5(uuid.NamespaceURL, actions.FullName+resultsAction.FullName).String(),
+		ConnectionRequirements: []map[string]string{{"collection": actions.CollectionDisplayName}},
+		Level:                  consts.LibFreeLevel,
+		SlugName:               text.GetSlugName(libraryAutomation.Name),
+		Rank:                   consts.LowSearchPriority,
+		Categories:             actions.LibraryCategories,
+		Vendors:                []string{actions.CollectionDisplayName},
+	}
+}
+
+func SetMetadataFields(libraryAutomation *types.Automation, actions, resultsAction *types.Action, md *types.Metadata, resultsActionDescription string) error {
+	setAutomationID(libraryAutomation, actions, resultsAction, md)
+	appendResultsActionsParams(resultsAction, md)
+
+	if err := sortVendorsAndCategories(md, libraryAutomation); err != nil {
+		return errors.Wrap(err, "failed to sort")
+	}
+
+	setMetadataBreakdown(actions, md, resultsActionDescription)
+
+	return nil
+}
+
+func setMetadataBreakdown(actions *types.Action, md *types.Metadata, resultsActionDescription string) {
+	breakdownString := fmt.Sprintf("1. %s with %s.\n", text.FixCapitalization(actions.DisplayName), actions.CollectionDisplayName)
+	breakdownString += fmt.Sprintf("2. %s\n", resultsActionDescription)
+	md.Breakdown = breakdownString
+}
+
+func sortVendorsAndCategories(md *types.Metadata, libraryAutomation *types.Automation) error {
+	// we categorize result actions which should be mentioned as vendors by whether their categories is non-empty
+	sort.Strings(md.Vendors)
+	if len(md.Categories) == 0 {
+		return errors.Errorf("%s is missing categories", libraryAutomation.Name)
+	}
```

Comment:
no point in sorting if in the next step we can return an error

```suggestion
	if len(md.Categories) == 0 {
		return errors.Errorf("%s is missing categories", libraryAutomation.Name)
	}
	
	sort.Strings(md.Vendors)
```